pkg/utils/node/node.go

```diff
@@ -83,3 +83,17 @@ func GetCondition(n *v1.Node, match v1.NodeConditionType) v1.NodeCondition {
 	}
 	return v1.NodeCondition{}
 }
+
+func PatchTaints(ctx context.Context, kubeClient client.Client, oldNode, newNode *v1.Node) error {
+	// Strip the resource version off the original node. This ensures the patch will fail with a conflict if the node
+	// had been updated between the original GET and the PATCH. This is necessary because .spec.taints does not specify
+	// a patchMergeKey and patchStrategy. Ref: https://github.com/kubernetes/kubernetes/pull/113136
+	oldNodeStripped := oldNode.DeepCopy()
+	oldNodeStripped.ResourceVersion = ""
+
+	newTaints := newNode.Spec.Taints
+	newNodeCloned := oldNode.DeepCopy()
+	newNodeCloned.Spec.Taints = newTaints
+
+	return kubeClient.Patch(ctx, newNodeCloned, client.StrategicMergeFrom(oldNodeStripped))
```

Comment:
I question whether we should take in this hack since it takes someone understanding what's going on here and why this works. What's the issue that we think that we might encounter here by simplifying with an Update() call

Reply:
Is there some option that we can ask controller-runtime to support? Or is there a wrapper around our client that we should add that would support `PatchWithConflict()`

Reply:
That was the intention of wrapping it here, and calling out the hack in the comment. IIUC the main advantage is to reduce potential retries. I think we can probably start with Update and add this as an optimization down the road if necessary.